plugins/modules/inventory_intent.py

```diff
@@ -2989,40 +2929,42 @@ def get_diff_merged(self, config):
                 raise Exception(error_message)
 
         if self.config[0].get('add_user_defined_field'):
-            field_name = self.config[0].get('add_user_defined_field').get('name')
+            udf_field_list = self.config[0].get('add_user_defined_field')
 
-            if field_name is None:
-                self.status = "failed"
-                self.msg = "Mandatory paramter for User Define Field 'name' is missing"
-                self.log(self.msg, "ERROR")
-                self.result['response'] = self.msg
-                return self
+            for udf in udf_field_list:
+                field_name = udf.get('name')
 
-            # Check if the Global User defined field exist if not then create it with given field name
-            udf_exist = self.is_udf_exist(field_name)
+                if field_name is None:
+                    self.status = "failed"
+                    self.msg = "Error: The mandatory parameter 'name' for the User Defined Field is missing. Please provide the required information."
+                    self.log(self.msg, "ERROR")
+                    self.result['response'] = self.msg
+                    return self
 
-            if not udf_exist:
-                # Create the Global UDF
-                self.create_user_defined_field().check_return_status()
+                # Check if the Global User defined field exist if not then create it with given field name
+                udf_exist = self.is_udf_exist(field_name)
 
-            # Get device Id based on config priority
-            device_ips = self.get_device_ips_from_config_priority()
-            device_ids = self.get_device_ids(device_ips)
+                if not udf_exist:
+                    # Create the Global UDF
+                    self.create_user_defined_field(udf).check_return_status()
 
-            if not device_ids:
-                self.status = "failed"
-                self.msg = "Can't Assign Global User Defined Field to device as device's are not present in Cisco Catalyst Center"
-                self.result['changed'] = False
-                self.result['response'] = self.msg
-                self.log(self.msg, "INFO")
-                return self
+                # Get device Id based on config priority
+                device_ips = self.get_device_ips_from_config_priority()
+                device_ids = self.get_device_ids(device_ips)
+
+                if len(device_ids) == 0:
+                    self.status = "failed"
+                    self.msg = "Can't Assign Global User Defined Field to device as device's are not present in Cisco Catalyst Center"
```

Comment:
self.msg = "Unable to assign Global User Defined Field: No devices found in Cisco Catalyst Center. Please add devices to proceed."

Reply:
added.